test/extended/builds/new_app.go

```diff
@@ -0,0 +1,40 @@
+package builds
+
+import (
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[builds][Conformance] oc new-app", func() {
+	// Previously, the maximum length of app names creatable by new-app has
+	// inadvertently been decreased, e.g. by creating an annotation somewhere
+	// whose name itself includes the app name.  Ensure we can create and deploy
+	// an app with a 58 character name [63 maximum - len('-9999' suffix)].
+
+	oc := exutil.NewCLI("new-app", exutil.KubeConfigPath())
+
+	g.JustBeforeEach(func() {
+		g.By("waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.It("should succeed with a --name of 58 characters", func() {
```

Comment:
there was some earlier discussion of this (I made the same suggestion), @jim-minter had a good justification for doing it as an extended test but unfortunately i went and merged this before making sure he updated the test code w/ a comment explaining that justification.